src/progress.rs

```diff
@@ -512,6 +512,21 @@ impl Progress {
     pub fn pause(&mut self) {
         self.paused = true;
     }
+
+    /// Update inflight msgs and next_idx
+    pub fn update_state(&mut self, last: u64) {
+        match self.state {
+            ProgressState::Replicate => {
+                self.optimistic_update(last);
+                self.ins.add(last);
+            }
+            ProgressState::Probe => self.pause(),
+            _ => panic!(
```

Comment:
```suggestion
            ProgressState::Snapshot => panic!(
```

There's no benefit to using `_` here, the only other option is `ProgressState::Snapshot`. Can you change it to be explicit?

I also worry that it's not particularly appropriate to panic in this state... however that's outside the scope of this PR. So let's not worry about that here. :)